src/hotspot/share/prims/jni.cpp

```diff
@@ -2055,8 +2055,11 @@ JNI_ENTRY(jfieldID, jni_GetStaticFieldID(JNIEnv *env, jclass clazz,
   k->initialize(CHECK_NULL);
 
   fieldDescriptor fd;
+  // strict fields need special tracking during <clinit>; do not hand them out so early
   if (!k->is_instance_klass() ||
-      !InstanceKlass::cast(k)->find_field(fieldname, signame, true, &fd)) {
+      !InstanceKlass::cast(k)->find_field(fieldname, signame, true, &fd) ||
+      // strict fields need special tracking during <clinit>; do not hand them out so early
+      (fd.access_flags().is_strict() && !InstanceKlass::cast(k)->is_initialized())) {
```

Comment:
Should this be an assert rather than a runtime check given that we force the class to be initialized a few lines above?